hbase-server/src/main/java/org/apache/hadoop/hbase/io/hfile/HFileWriterImpl.java

```diff
@@ -319,6 +323,9 @@ protected void checkBlockBoundary() throws IOException {
       shouldFinishBlock = blockWriter.encodedBlockSizeWritten() >= hFileContext.getBlocksize()
         || blockWriter.blockSizeWritten() >= hFileContext.getBlocksize();
     }
+    if (blockWriter.isSizeLimitCompressed()) {
+      shouldFinishBlock &= blockWriter.shouldFinishBlock();
```

Comment:
related to the above comment by @bbeaudreault 

in what situation the shouldFinishBlock = true and blockWriter.shouldFinishBlock() = false ? is it possible?

Reply:
Yes, shouldFinishBlock could be true at this point because so far here we just checked "raw" uncompressed size or encoded uncompressed against BLOCK_SIZE. It is possible that these sizes are higher than BLOCK_SIZE, but the compressed size might still be less than the BLOCK_SIZE.